### ISSUE TITLE
Strange Lr

### ISSUE THREAD

**Issue by huixiancheng** [placeforyiming/IROS21-FIDNet-SemanticKITTI#4]
Compare with lr_policy 2(Top), lr_policy(Bottom) 1 look like very strange. Kind of like multi warm_up.
With lr_policy 1, the loss will change jumping up and down between epochs. 
- lr_policy 2:[0.0004, 0.0004, 0.0008, 0.0008, 0.0012, 0.0012, 0.0016, 0.0016, 0.002, 0.002, 0.0018, 0.0018, 0.0016, 0.0016, 0.0014, 0.0014, 0.0012, 0.0012, 0.001, 0.001, 0.0008, 0.0008, 0.0006, 0.0006, 0.0004, 0.0004]

- lr_policy 1:[0.0004, 0.0004, 0.0008, 0.0008, 0.0012, 0.0012, 0.0016, 0.0016, 0.002, 0.002, 0.0016, 0.0016, 0.0012, 0.0012, 0.0008, 0.0008, 0.0012, 0.0012, 0.0006, 0.0006, 0.0008, 0.0008, 0.0004, 0.0004, 0.0002, 0.0002]

![myplot1](https://user-images.githubusercontent.com/65054204/134629534-1452c0cc-f4dd-4200-897a-556d0b1dd62b.png)
![myplot2](https://user-images.githubusercontent.com/65054204/134629543-47844099-c498-4d63-b6a5-3a0c9a56e1ed.png)

**Comment by placeforyiming** [comment]
> Compare with lr_policy 2(Top), lr_policy(Bottom) 1 look like very strange. Kind of like multi warm_up.
> With lr_policy 1, the loss will change jumping up and down between epochs.
> 
> * lr_policy 2:[0.0004, 0.0004, 0.0008, 0.0008, 0.0012, 0.0012, 0.0016, 0.0016, 0.002, 0.002, 0.0018, 0.0018, 0.0016, 0.0016, 0.0014, 0.0014, 0.0012, 0.0012, 0.001, 0.001, 0.0008, 0.0008, 0.0006, 0.0006, 0.0004, 0.0004]
> * lr_policy 1:[0.0004, 0.0004, 0.0008, 0.0008, 0.0012, 0.0012, 0.0016, 0.0016, 0.002, 0.002, 0.0016, 0.0016, 0.0012, 0.0012, 0.0008, 0.0008, 0.0012, 0.0012, 0.0006, 0.0006, 0.0008, 0.0008, 0.0004, 0.0004, 0.0002, 0.0002]
> 
> ![myplot1](https://user-images.githubusercontent.com/65054204/134629534-1452c0cc-f4dd-4200-897a-556d0b1dd62b.png)
> ![myplot2](https://user-images.githubusercontent.com/65054204/134629543-47844099-c498-4d63-b6a5-3a0c9a56e1ed.png)

The second one is supposed to be the cyclical learning rate scheduler, the same as the SalsaNext. But it means a very long training procedure waiting for the best checkpoint. So I use the current 'shorter' one, that makes the training finished in a day.
If you have enough GPUs, you can replace it with a standard cyclical learning rate scheduler, and implement a simple evaluation function only on range images. You should can get a better checkpoint in that way.

**Comment by huixiancheng** [comment]
Thanks a lot. 🙇‍♂️  Have done it to train peer epoch with val. More, have your ever try use sgd instead of adamw?

**Comment by placeforyiming** [comment]
> Thanks a lot. 🙇‍♂️ Have done it to train peer epoch with val. More, have your ever try use sgd instead of adamw?

No, I didn't try SGD. SGD usually has better performance in general CV tasks, but it's hard to find the best settings.  So it needs some extra experiments with gpu cost

**Comment by huixiancheng** [comment]
thx a lot for your kindly reply
